src/foam/u2/layout/Card.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.layout',
+  name: 'Card',
+  extends: 'foam.u2.borders.CardBorder',
+  documentation: `
+    A card based on a responsive grid system
+  `,
+
+  imports: [
+    'displayWidth?'
+  ],
+
+  css: `
+    ^ {
+      margin: 8px;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'cols',
+      documentation: `
+        Sets up a standard default column width across all display types
+      `,
+      value: 12
+    },
+    {
+      class: 'Int',
+      name: 'xxsCols',
+      documentation: `
+        The column width for the smaller end of smartphone devices and smartphone portrait screens:
+        Max-width @ 320px and an 8 column grid
+      `,
+      expression: function(cols) {
+        return cols > 8 ? 8 : cols;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'xsCols',
+      documentation: `
+        The column width for the regular end of smartphone devices
+        Max-width @ 576px, min-width @ 320px and an 8 column grid
+      `,
+      expression: function(cols) {
+        return cols > 8 ? 8 : cols;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'smCols',
+      documentation: `
+        The column width for the larger end of smartphone devices and landscape smartphone screens:
+        Max-width @ 768px, min-width @ 576px and a 12 column grid
+      `,
+      expression: function(cols) {
+        return cols;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'mdCols',
+      documentation: `
+        The column width for most tablet screens and portrait tablet screens:
+        Max-width @ 960px, min-width @ 768px and a 12 column grid
+      `,
+      expression: function(cols) {
+        return cols;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'lgCols',
+      documentation: `
+        The column width for the smaller end of desktop screens:
+        Max-width @ 1280px, min-width @ 960px and a 12 column grid
+      `,
+      expression: function(cols) {
+        return cols;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'xlCols',
+      documentation: `
+        The column width for the majority of desktop screens:
+        Max-width @ 1440px, min-width @ 1280px and a 12 column grid
+      `,
+      expression: function(cols) {
+        return cols;
+      }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this.SUPER();
+      this.addClass(this.myClass());
+
+      // set the default value the xl if display width doesn't exist
+      let flex;
+
+      if ( this.displayWidth ) {
```

Comment:
displayWidth can change so you want to be bound to it.
Also, you can enumerate enums so this kinda thing could be reduced a little by using array functions on the enum values but I think there's a better way to reduce it:
```
this.style({
  'flex-grow': this.displayWidth$.map(dw => {
    dw = dw || foam.u2.layout.DisplayWidth.XL;
    flex = this[dw.name.toLowerCase() + 'Cols'] / dw.cols;
  })
})
```
That said, there's something that's not quite sitting well with me here because I'm worried we're going to have to add a bunch of properties to every view that cares about the display width and that can be kind of annoying but maybe we won't actually be doing that as frequently as I think. Let's keep an eye out for us needing to do this frequently and if it becomes a problem we'll maybe turn this sort of thing into a map or maybe even a more formal modelled object.

Reply:
Done but factor doesn't fire-off upon resizing